### ISSUE TITLE
Fix S3874 FP: Should not be raised on Deconstruct methods

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#5816]
### Description

S3874 `"out" and "ref" parameters should not be used` is raised for out parameters for custom Deconstruct methods. The C# spec defines the signatures for `Deconstruct` methods and requires `out` parameters [docs](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/deconstruct#user-defined-types).

### Repro steps

Create Deconstruct method as either instance, or extension method:
```cs
public void Deconstruct(out string fname, out string mname, out string name)

public static void Deconstruct(this PropertyInfo p, out bool isStatic, out bool isReadOnly, out bool isIndexed, out Type propertyType)
```

### Expected behavior

Don't raise S3874 on out parameters of Deconstruct methods.

### Actual behavior

S3874 is raised.

### Known workarounds

None (Besides suppression)

### Related information
